src/docs/release/breaking-changes/text-field-material-localizations.md

```diff
@@ -0,0 +1,140 @@
+by the material library.
+To introduce a MaterialLocalizations, either use a MaterialApp at the root of your application to
+include them automatically, or add a Localization widget with a MaterialLocalizations delegate.
+The specific widget that could not find a MaterialLocalizations ancestor was:
+  TextField
+```
+
+## Context
+
+If your `TextField` is within the scope of a `MaterialApp`, the
+`DefaultMaterialLocalizations` will already be instantiated and may suit your needs without having
+to make a change to your existing code.
+
+If your `TextField` is not within a `MaterialApp`, you may provide the localizations of
+your choosing using the `Localizations` widget.
+
+This change will initially be introduced with an opt-in flag
+`TextField.canAssertMaterialLocalizations`, to allow developers time to migrate
+without experiencing broken code. This temporary opt-in flag will be removed in a follow up change.
+
+## Migration guide
+
+If you are seeing an assertion error, make sure locale information is being
+provided to your `TextField`.
+
+Code before migration:
+
+```dart
+import 'package:flutter/material.dart';
+void main() => runApp(Foo());
+class Foo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MediaQuery(
+      data: const MediaQueryData(),
+      child: Directionality(
+        textDirection: TextDirection.ltr,
+        child: TextField(),
+      ),
+    );
+  }
+}
+```
+
+Code after migration (Providing localizations via the `MaterialApp`):
+
+```dart
+import 'package:flutter/material.dart';
+void main() => runApp(Foo());
+class Foo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: TextField(),
+    );
+  }
+}
+```
+
+Code after migration (Providing localizations via the `Localizations` widget):
+
+```dart
+import 'package:flutter/material.dart';
+void main() => runApp(Foo());
+class Foo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Localizations(
+      locale: const Locale('en', 'US'),
+      delegates: <LocalizationsDelegate<dynamic>>[
+        DefaultWidgetsLocalizations.delegate,
+        DefaultMaterialLocalizations.delegate,
+      ],
+      child: MediaQuery(
+        data: const MediaQueryData(),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: TextField(),
+        ),
+      ),
+    );
+  }
+}
+```
+
+Code to validate migration:
+
+```dart
+import 'package:flutter/material.dart';
+void main() => runApp(Foo());
+class Foo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: TextField(canAssertMaterialLocalizations: true),
+    );
+  }
+}
+```
+
+## Timeline
+
+TBD - Add the version number that contains this change.
+
+## References
+
+API documentation:
+* [`TextField`][]
```

Comment:
Do any of these API links rely on docs newer than 1.17 stable? If so, link on the master channel.